### PR TITLE
feat: tshirt registration, scores routes

### PR DESCRIPTION
**Pull request by vigneshd332**
### Changes

- [x] Add Scores Route
- [x] Add T-Shirt Registration Route
- [x] Add Tests

### CODE REVIEW

File: server/routers/tshirt.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+async def register_tshirt(
+    size_requested: TshirtRequestModel,
+    token: str = Depends(JWTBearer()),
+    database: Session = Depends(get_database),
+) -> None:
+    """
+    POST route for Tshirt Registration
+    """
+    try:
+        email = decode_jwt(token)["email"]
```
Comment (Muhesh7) [inline server/routers/tshirt.py]: "email" ?
it should be "user_email"

it should have shown u internal server error(if u ran and checked it atleast once ofcourse).

File: server/routers/tshirt.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        user_id = database.query(Users.id).filter_by(email=email).first()
+        user_id = 1
```
Comment (indreshp135) [inline server/routers/tshirt.py]: 🤓

Comment (vigneshd332) [reply server/routers/tshirt.py]: Ah shit, that was for testing 😭 I'll change rn

File: server/routers/tshirt.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+            raise GenericError(
+                f"Invalid size requested: Size can only be {' '.join(valid_sizes)}"
+            )
+        # if not user_id:
+        #     raise GenericError("User not found")
+        tshirt = Tshirt(
+            size=size_requested.size,
+            user_id=user_id,
+        )
+        database.add(tshirt)
```
Comment (Muhesh7) [inline server/routers/tshirt.py:54]: There should be a check that allows a user to register onli once

File: server/routers/tshirt.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        database.commit()
+        database.close()
```
Comment (Muhesh7) [inline server/routers/tshirt.py]: Dependency injection takes care of the close()

File: server/routers/tshirt.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        return {"message": "Tshirt registered successfully"}
+    except GenericError as exception:
+        logger.error("Tshirt registration failed due to {exception}")
+        raise HTTPException(
+            status_code=500,
+            detail="An unexpected error occurred while registering tshirt",
+            headers={"X-Error": str(exception)},
+        ) from exception
```
Comment (Muhesh7) [inline server/routers/tshirt.py:67]: Write tests for this route

File: server/models/scores.py [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+        title="Cluster",
+        description="Cluster Name",
+    )
+    points: int = Field(
+        ...,
+        title="Points",
+        description="Points for the cluster",
+    )
+
+    class Config:
```
Comment (Muhesh7) [inline server/models/scores.py]: this class is not needed

**Review by Muhesh7 — CHANGES_REQUESTED** [review]

File: server/routers/scores.py [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+        for cluster in clusters:
+            cluster_list.update({cluster.id: cluster.name})
+        return get_all_scores(
+            dept_list=dept_list, cluster_list=cluster_list, points=points
+        )
+    except GenericError as exception:
+        logger.error(f"Fetching Events failed due to {exception}")
+        raise HTTPException(
+            status_code=403,
+            detail=f"An unexpected error occurred while fetching events:{exception}",
```
Comment (Muhesh7) [inline server/routers/scores.py:55]: avoid sending the actual exception to client.
change this in all the routes

**Review by Muhesh7 — APPROVED** [review]

File: server/routers/scores.py (hunk 7 above)
Comment (Muhesh7) [inline server/routers/scores.py:55]: Dont send the actual exception to client
change this in every other route

**Review by Muhesh7 — CHANGES_REQUESTED** [review]

**Review by Muhesh7 — APPROVED** [review]

**Review by indreshp135 — APPROVED** [review]

**Pull request merged**